src/main/java/org/sagebionetworks/web/client/security/AuthenticationControllerImpl.java

```diff
@@ -274,13 +281,29 @@ public static boolean isQuarantined(EmailQuarantineStatus status) {
     );
   }
 
+  public void clearLocalStorage() {
+    Map<String, String> storedKeyValues = new HashMap<String, String>();
```

Comment:
Store local storage keys that should persist in this map, clear local storage, and then repopulate